kafka/factories.go

```diff
@@ -0,0 +1,23 @@
+package kafka
+
+import (
+	"runtime"
+)
+
+func NewMessageProviderFactory(topicName string, props map[string]string, groupID string) MessageProviderFactory {
+	goarch := runtime.GOARCH
+	goos := runtime.GOOS
+	if goos == "darwin" && goarch == "arm64" {
```

Comment:
Even when using MacOS arm64 it will sometimes be necessary to run using the ConfluentMessageProviderFactory - e.g. when running correctness tests (The segment factory doesn't implement the Kafka rebalance protocol properly). Can we add someone of overridding the behaviour here? E.g. check for an environment variable and if present use the Confluent factory.

Reply:
This PR intended to make pranadb to decide kafka client on runtime, because I would like to add more options of client implementations (e.g. sarama, watermill `https://github.com/ThreeDotsLabs/watermill`).
Cureently pranadb choose its client type during build time by `//go:build...`. IMHO it might be only good for reducing binary size, but less flexibility and increase complexity of testing. 

Is it good idea  (based from this PR's idea) to add more BrokerClientType in `conf/conf.go`, something like `BrokerClientSarama` `BrokerClientWaterMill`?

Reply:
The only client library we want to use for real use right now is the Confluent client library as this is the only one which we know implements the Kafka rebalance protocol properly.

Reply:
I think as long as we allow an ENV_VAR override to select the confluent lib when running on MacOS arm64 we are good, as this will be needed for some tests.